src/common/transformations/src/transformations/common_optimizations/broadcast_transition.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/broadcast_transition.hpp"
+
+#include <memory>
+#include <openvino/opsets/opset1.hpp>
+#include <openvino/opsets/opset10.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+#include <vector>
+
+#include "itt.hpp"
+#include "transformations/utils/utils.hpp"
+
+ov::pass::BroadcastTransition::BroadcastTransition() {
+    MATCHER_SCOPE(BroadcastTransition);
+    auto bcast_data_m = pass::pattern::any_input();
+    auto target_shape_m = pass::pattern::any_input(pass::pattern::has_static_shape());
+    auto bcast_m = pass::pattern::wrap_type<opset1::Broadcast, opset10::Broadcast>({bcast_data_m, target_shape_m},
+                                                                                   pass::pattern::consumers_count(1));
+    auto eltwise_input_m = pass::pattern::any_input(pass::pattern::has_static_rank());
+    auto eltwise_1 = pass::pattern::wrap_type<op::util::BinaryElementwiseArithmetic>({eltwise_input_m, bcast_m});
+    auto eltwise_2 = pass::pattern::wrap_type<op::util::BinaryElementwiseArithmetic>({bcast_m, eltwise_input_m});
+    auto eltwise_m = std::make_shared<pass::pattern::op::Or>(OutputVector{eltwise_1, eltwise_2});
```

Comment:
Our pattern matcher knows which operations are commutative (l[ink](https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/pattern/matcher.cpp#L132)). However, here you also can work with non-commutative operations, like Subtract and Divide. So, this is just an observation.

Reply:
Good point, thanks. I used `Or` exactly to cover non-commutative operations in the pattern. As for me, this is the shortest way to handle both cases.